src/Controls/tests/Core.UnitTests/Layouts/GridLayoutTests.cs

```diff
@@ -184,5 +186,45 @@ public void ColumnDefinitionsGetBindingContext()
 
 			Assert.Equal(def2.BindingContext, context);
 		}
+
+		[Fact]
+		public async Task ColumnDefinitionDoesNotLeak()
+		{
+			// Long-lived column, like from a Style in App.Resources
+			var column = new ColumnDefinition();
+			WeakReference reference;
+
+			{
+				var grid = new Grid();
+				grid.ColumnDefinitions.Add(column);
+				reference = new(grid);
+			}
+
+			await Task.Yield();
+			GC.Collect();
+			GC.WaitForPendingFinalizers();
+
+			Assert.False(reference.IsAlive, "Grid should not be alive!");
```

Comment:
Can you help me understand what's going on behind the scenes here? What are the "pending finalizers" in WaitForPendingFinalizers?

Reply:
Because the GC is not deterministic... It can be hard to write a test to check for a memory leak.

Through trial and error, I've found this works reliably (at least in .NET 6+):

* `await Task.Yield()` - a simple way to get the second half of the test on a different thread
* `GC.Collect()` - runs the GC, but it's not 100% what it decides actually clean up.
* `GC.WaitForPendingFinalizers()`, waits on any finalizers like `~` you can define in classes. It's possible we could remove this call and the test still pass.

All this work is to force the GC to run and cleanup `var grid`, and assert it is gone at the end.

In the past, though... We had to do weird things like this for Xamarin apps:

* https://github.com/xamarin/java.interop/commit/220b87fbbb8d986c2c8951757aa3ea45d8e7e7eb

Reply:
gotcha, thanks for elaborating!